source/driver/Driver.cpp

```diff
@@ -593,7 +593,7 @@ bool Driver::parseAllSources() {
 
         auto findMissingNames = [&](const std::shared_ptr<SyntaxTree>& tree,
                                     flat_hash_set<std::string_view>& missing) {
-            auto& meta = tree->getMetadata();
+            const auto& meta = ParserMetadata::fromSyntax(tree->root());
```

Comment:
You should not recreate the metadata here, it's an expensive operation. The parser already builds up the metadata incrementally, you should modify the parser to also include your new field.